### PR TITLE
Fix thatch bed drops & ice pile rendering (1.18)

### PR DESCRIPTION
**Pull request by Charnuz**
Before:
before-ice-fix
After:
after-ice-fix

### CODE REVIEW

File: src/main/java/net/dries007/tfc/common/blocks/TFCBlocks.java [hunk 1]
```diff
@@ -257,7 +256,7 @@
     // Misc
 
     public static final RegistryObject<Block> THATCH = register("thatch", () -> new ThatchBlock(ExtendedProperties.of(Properties.of(Material.PLANT).strength(0.6F, 0.4F).noOcclusion().sound(TFCSounds.THATCH)).flammable(50, 100)), MISC);
-    public static final RegistryObject<Block> THATCH_BED = register("thatch_bed", () -> new ThatchBedBlock(Properties.of(Material.REPLACEABLE_PLANT).sound(TFCSounds.THATCH).strength(0.6F, 0.4F)));
+    public static final RegistryObject<Block> THATCH_BED = register("thatch_bed", () -> new ThatchBedBlock(ExtendedProperties.of(Properties.of(Material.REPLACEABLE_PLANT).sound(TFCSounds.THATCH).strength(0.6F, 0.4F)).blockEntity(TFCBlockEntities.THATCH_BED)));
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/common/blocks/TFCBlocks.java]: flammable?

File: src/main/java/net/dries007/tfc/mixin/IceBlockMixin.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    /**
+     * Makes ice blocks stop rendering the faces of connected ice piles.
+     */
+    @Override
+    public boolean skipRendering(BlockState state, BlockState otherState, Direction facing)
+    {
+        return Helpers.isBlock(otherState, TFCBlocks.ICE_PILE.get()) || super.skipRendering(state, otherState, facing);
+    }
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/mixin/IceBlockMixin.java]: Can you inject at head here, set to true and cancel instead? Overrides should be avoided

File: src/main/java/net/dries007/tfc/util/InteractionManager.java [hunk 3]
```diff
@@ -74,19 +72,21 @@ public static void registerDefaultInteractions()
...
+                    final BlockState headState = world.getBlockState(headPos);
+                    if (Helpers.isBlock(baseState, TFCTags.Blocks.THATCH_BED_THATCH) && Helpers.isBlock(headState, TFCTags.Blocks.THATCH_BED_THATCH))
                     {
                         final BlockPos playerPos = player.blockPosition();
                         if (playerPos != headPos && playerPos != basePos)
                         {
                             world.setBlock(basePos, bed.setValue(ThatchBedBlock.PART, BedPart.FOOT).setValue(ThatchBedBlock.FACING, direction), 18);
                             world.setBlock(headPos, bed.setValue(ThatchBedBlock.PART, BedPart.HEAD).setValue(ThatchBedBlock.FACING, direction.getOpposite()), 18);
-                            stack.shrink(1);
+                            world.getBlockEntity(headPos, TFCBlockEntities.THATCH_BED.get()).ifPresent(entity -> entity.setBed(headState, baseState, stack.split(1)));
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/util/InteractionManager.java]: may as well change this param to level while you are here

**Review by eerussianguy — CHANGES_REQUESTED** [review]

File: src/main/java/net/dries007/tfc/TerraFirmaCraft.java [hunk 4]
```diff
@@ -93,14 +92,20 @@ public TerraFirmaCraft()
         TFCContainerTypes.CONTAINERS.register(bus);
         TFCEntities.ENTITIES.register(bus);
         TFCFluids.FLUIDS.register(bus);
+        TFCRecipeTypes.RECIPE_TYPES.register(bus);
         TFCRecipeSerializers.RECIPE_SERIALIZERS.register(bus);
         TFCSounds.SOUNDS.register(bus);
         TFCParticles.PARTICLE_TYPES.register(bus);
         TFCBlockEntities.TILE_ENTITIES.register(bus);
+        TFCLoot.registerAll(bus);
```
Comment (Charnuz) [inline src/main/java/net/dries007/tfc/TerraFirmaCraft.java:100]: Can register each registry individually if preferred

**Review by eerussianguy — APPROVED** [review]

**Review by alcatrazEscapee — APPROVED** [review]

**Pull request merged**